### PR TITLE
Converting hologram to drf serializers then generating the open api s…

### PR DESCRIPTION
**Pull request by sidpatel13**
instead of converting hologram straight to the open api spec, we convert the serializers to drf serializers and then use those in conjunction with drf spectacular to generate the open api schema. drf spectacular, the package we use to generate the docs tends to understand the drf serializers a lot better.

Checkout the changes using this Loom [link](https://www.loom.com/share/0d8f43dd79ae48ccb4588907f24e2c3b)

### CODE REVIEW

**Comment by sidpatel13** [comment]
Added a loom[ link](https://www.loom.com/share/0d8f43dd79ae48ccb4588907f24e2c3b) showing what got changed as a part of this update

File: openapi-v2.yaml [hunk 1]
```diff
@@ -2,12 +2,11 @@ openapi: 3.0.3
...
-  description: "# How to use this API\n\nThe dbt Cloud API is intended for enqueuing\
-    \ runs from a job, polling for run progress,\nand downloading artifacts after\
-    \ jobs have completed running. Operational endpoints around\ncreating, modifying,\
-    \ and deleting _objects_ in dbt Cloud are still in flux. These endpoints\nare\
-    \ largely undocumented in API v2.\n\nThe API docs are generated from an openapi\
-    \ spec defined in the\n[dbt-cloud-openapi-spec](https://github.com/dbt-labs/dbt-cloud-openapi-spec)\n\
+  description: "# How to use this API\n\nThe dbt Cloud API v2 contains endpoints for\
+    \ enqueuing runs from a job, polling for run progress,\nand downloading artifacts\
+    \ after jobs have completed running. For operational endpoints to\ncreate, modify,\
+    \ and delete _objects_ in dbt Cloud, consider using [API v3](/dbt-cloud/api-v3#/)\n\
```
Comment (GideonShils) [inline openapi-v2.yaml:8]: I'm pretty sure this link won't work. Can we update in dbt-cloud to be the full URL?

**Review by MitchellBarker — COMMENTED** [review]
Down to approve once this [PR](https://github.com/dbt-labs/dbt-cloud/pull/8756) goes through (know you just tagged for viz -> and I'm really just commenting so I can drop it from my gh view haha)

**Comment by sidpatel13** [comment]
https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/dbt-labs/dbt-cloud-openapi-spec/7a72ecdd8e29107954ec68e3f0987096baccb26e/openapi-v2yaml#/
https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/dbt-labs/dbt-cloud-openapi-spec/7a72ecdd8e29107954ec68e3f0987096baccb26e/openapi-v3.yaml#/

**Review by MitchellBarker — APPROVED** [review]
🥳

**Pull request merged**